### PR TITLE
[DDW-480] Tile view in the delegation center

### PR DESCRIPTION
**Pull request by DeeJayElly**
This PR implements tile view in Delegation center for delegated stake pools with ranking and saturation for stake pools.

Design: https://zpl.io/bLyl0YQ

Requirements: https://docs.google.com/document/d/1Zv7mDFe5pf3nsW-yUc1_7shaEXlsRaHpOZ_oh4wR4vo/edit?usp=sharing 

## Screenshots

Screen Shot 2021-01-14 at 6 43 25 AM

Screen Shot 2021-01-14 at 6 43 09 AM

Screen Shot 2021-01-14 at 6 43 02 AM

Screen Shot 2021-01-14 at 6 42 55 AM

Screen Shot 2021-01-14 at 6 42 43 AM

Screen Shot 2021-01-14 at 6 42 29 AM

Screen Shot 2021-01-14 at 6 42 18 AM

Screen Shot 2021-01-14 at 6 42 02 AM

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1607894891032800)
- [ ] Test use cases based on the requirements : https://docs.google.com/document/d/1Zv7mDFe5pf3nsW-yUc1_7shaEXlsRaHpOZ_oh4wR4vo/edit?usp=sharing
- [ ] Test if the component UI matches the design based on https://zpl.io/bLyl0YQ

## Test Cases

Additional document for testing: [States table](https://docs.google.com/document/d/1XK4anL8RgkA3QiTurqXvDu_x0D_Rriu_7mun7ySR19I/edit)

**Scenario 1: Viewing the tile view in the Delegation Center**
Given that I am on Daedalus
When I select Stake Pool's Delegation Center
Then I am shown a tile view of wallets and pools they are delegating to
And the display in epoch, epoch+1 or epoch+2 will be either
||
|--|
|<pool_ticker>|
|UNKNOWN|
|UNDELEGATED|

**Scenario 2: Re-delegating a wallet to another pool**
Given that I am on Daedalus Delegation Center
And I have a wallet delegated to a pool
And the pool is in epoch+2
When I hover over the pool
Then the cursor is changed to a “hand”
And 'REDELEGATE' appears at bottom
When I select 'REDELEGATE'
Then I am led to the delegation wizard

**Scenario 3: Delegating an undelegated wallet to a pool**
Given that I am on Daedalus Delegation Center
And I have a wallet not delegated to any pool in epoch+2
And  “UNDELEGATED” is shown
When I hover over the pool
Then I am able to click on the 'DELEGATE' button
And I am led to the delegation wizard

**Scenario 4: Viewing the rank of a pool**
Given that I am on Daedalus Delegation Center
When one of my wallets is delegated to a certain pool
And the pool is in epoch+2
Then I can see the pool ranking number within the square that shows the pool

**Scenario 5: Viewing the saturation level of a pool**
Given that I am on Daedalus Delegation Center
When one of my wallets is delegated to a certain pool
And the pool is in epoch+2
Then I can see the saturation level in form of a coloured line above the pool ranking

**Scenario 6: Unknown pool (i.e. filtered out by SMASH)**
Given that I am on Daedalus Delegation Center
When a pool that I am delegating to is marked as 'UNKNOWN' by being filtered by SMASH server
Then no information is shown upon hovering on that pool

**Scenario 7: Hovering on Epoch+1 pool**
Given that I am in the Daedalus Delegation Center
And there is a pool at epoch+1
When I hover on the pool
Then no information is shown

**Scenario 8: Delegating from an unknown pool (i.e. filtered out by SMASH)**
Given that I am on Daedalus Delegation Center
And there is an unknown pool at epoch+2
When I hover over the pool
Then the cursor is changed to a “hand”
And 'REDELEGATE' appears at bottom
When I select 'REDELEGATE'
Then I am led to the delegation wizard

**Scenario 9:Wallet that has never been delegated**
Given that I am on Daedalus Delegation Center
And I have a wallet that has never been delegated
When I observe the tiles
Then I see the following displayed
|Epoch|Epoch+1|Epoch+2|
|--|--|--|
|Undelegated|Undelegated|Undelegated|

**Scenario 10:No pending changes, currently delegated to X**
Given that I am on Daedalus Delegation Center
And I have a wallet that has been delegated to Pool X
And the wallet has never been redelegated
And the wallet is delegated to Pool X in current epoch
When I observe the tiles
Then I see the following displayed
|Epoch|Epoch+1|Epoch+2|
|--|--|--|
|X|X|X|

**Scenario 11:No pending changes, currently delegated to X, then redelegate to Y**
Given that I am on Daedalus Delegation Center
And I have a wallet that has been delegated to Pool X
And the wallet has never been redelegated
And the wallet is delegated to Pool X in current epoch
When I redelegate to Pool Y
Then I see the following displayed
|Epoch|Epoch+1|Epoch+2|
|--|--|--|
|X|X|Y|

**Scenario 12:Currently delegated to X, 1 pending change from epoch + 1**
Given that I am on Daedalus Delegation Center
And I have a wallet that has been delegated to Pool X
And the wallet is delegated to Pool X in current epoch
And in the last epoch I have redelegated the wallet to Pool Y
When I observe the tiles
Then I see the following displayed
|Epoch|Epoch+1|Epoch+2|
|--|--|--|
|X|Y|Y|

**Scenario 13:Currently delegated to X, 2 pending changes from epoch + 1 to Pool Y, and from epoch + 2 to Pool Z**
Given that I am on Daedalus Delegation Center
And I have a wallet that has been delegated to Pool X
And the wallet is delegated to Pool X in current epoch
And in the last epoch I have redelegated the wallet to Pool Y
When I redelegate to Pool Z
Then I see the following displayed
|Epoch|Epoch+1|Epoch+2|
|--|--|--|
|X|Y|Z|

**Scenario 14:Currently delegated to X, 2 pending changes from epoch + 1 to Pool Y, and from epoch + 2 to Pool X**
Given that I am on Daedalus Delegation Center
And I have a wallet that has been delegated to Pool X
And the wallet is delegated to Pool X in current epoch
And in the last epoch I have redelegated the wallet to Pool Y
When I redelegate to Pool X
Then I see the following displayed
|Epoch|Epoch+1|Epoch+2|
|--|--|--|
|X|Y|X|

## Test Summary

Test results for build 15866
|Test cases | Results|Note|
|-----------|----|----|
|Scenario 1 | Passed||
|Scenario 2| Passed||
|Scenario 3| Passed||
|Scenario 4| Passed||
|Scenario 5| Passed||
|Scenario 6| Passed||
|Scenario 7| Passed||
|Scenario 8| Passed||
|Scenario 9| Passed||
|Scenario 10| Passed||
|Scenario 11| Passed||
|Scenario 12| Passed||
|Scenario 13| Passed||
|Scenario 14|Passed||

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by mioriohk** [comment]
@DeeJayElly , for build 15625. Here is the current report (Note: I have pasted Scenario list on top of this PR for ease of reference + modified where necessary):

|Test cases | Results|Note|
|-----------|----|----|
|Scenario 1 | Fail|No Stake pools displayed but 'formatting error' [See screenshot Scenario 1](https://drive.google.com/file/d/1nKZMx2OVEnqMRPjRVo-Af-1WeAQvhz4A/view?usp=sharing) |
|Scenario 2| Pass|I am finding it a bit confusing when a wallet which is currently being delegated to a pool, but on its epoch+2, it says “Undelegated” and on hover it displays “Redelegate”. From user point of view, it is as if I am redelegating an undelegated pool. Shouldn’t the current pool where the wallet is delegated to get displayed? [See screenshot Scenario 2](https://drive.google.com/file/d/1lXufWia0LEqL1wW2BHvmLPsi48VqD40-/view?usp=sharing) |
|Scenario 3| Pass||
|Scenario 4| Fail| Unable to see pool ranking due to error in Scenario 1|
|Scenario 5| Fail|Unable to see saturation due to error in Scenario 1|
|Scenario 6| Pending| Not tested yet, due to multiple issues found above|

**Comment by mioriohk** [comment]
@DeeJayElly , for build 15632
|Test cases | Results|Note|
|-----------|----|----|
|Scenario 1 | Pass| |
|Scenario 2| Fail|I am finding it a bit confusing when a wallet which is currently being delegated to a pool, but on its epoch+2, it says “Undelegated” and on hover it displays “Redelegate”. From user point of view, it is as if I am redelegating an undelegated pool. Shouldn’t the current pool where the wallet is delegated to get displayed? [See screenshot Scenario 2](https://drive.google.com/file/d/1lXufWia0LEqL1wW2BHvmLPsi48VqD40-/view?usp=sharing) |
|Scenario 3| Pass||
|Scenario 4| Pass||
|Scenario 5| Pass||
|Scenario 6| Pass||
|Scenario 7| Pass||

**Comment by gnpf** [comment]
@DeeJayElly For 15632 these are my findings:

**New:**
- Only when clicking the word "Delegate/redelegate" the dialog is displayed. I think the rest of the button should trigger this action too. Check [this recording](https://drive.google.com/file/d/17ddxqzyBvlwQ-dVBYSxPl4fAHcSI8wx4/view?usp=sharing).
- The color of the ranking doesn't match the displayed on the stake pools view. Check how it looks on [Delegation Center](https://drive.google.com/file/d/1Q8CfhUibJJhn1hYkA2a43vZaUqNeBAgn/view?usp=sharing) compared to [Stake pools screen.](https://drive.google.com/file/d/1BHmm0n42knKr6KecmwvrrhJjO3xRHzTm/view?usp=sharing)

**Comment by DeeJayElly** [comment]
> @DeeJayElly For 15632 these are my findings:
> 
> **New:**
> 
> * Only when clicking the word "Delegate/redelegate" the dialog is displayed. I think the rest of the button should trigger this action too. Check [this recording](https://drive.google.com/file/d/17ddxqzyBvlwQ-dVBYSxPl4fAHcSI8wx4/view?usp=sharing).
> * The color of the ranking doesn't match the displayed on the stake pools view. Check how it looks on [Delegation Center](https://drive.google.com/file/d/1Q8CfhUibJJhn1hYkA2a43vZaUqNeBAgn/view?usp=sharing) compared to [Stake pools screen.](https://drive.google.com/file/d/1BHmm0n42knKr6KecmwvrrhJjO3xRHzTm/view?usp=sharing)

@gnpf fixed

**Comment by DeeJayElly** [comment]
> @DeeJayElly , for build 15632. Only 1 failure. For complete results, please see Test Summary
> 
> Test cases	Results	Note
> Scenario 2	Fail	I am finding it a bit confusing when a wallet which is currently being delegated to a pool, but on its epoch+2, it says “Undelegated” and on hover it displays “Redelegate”. From user point of view, it is as if I am redelegating an undelegated pool. Shouldn’t the current pool where the wallet is delegated to get displayed? [See screenshot Scenario 2](https://drive.google.com/file/d/1lXufWia0LEqL1wW2BHvmLPsi48VqD40-/view?usp=sharing)
> Scenario 8	Pending	Aleksander to check with Nikola and Danilo how to get unknown pool in epoch+2

@mioriohk pls recheck Scenario 2, it should be covered now.

**Comment by nikolaglumac** [comment]
@yakovkaravelov please review this one 🙏

**Comment by mioriohk** [comment]
@DeeJayElly , for build 15640. The same issue as build 15632
|-----------|----|----|
|Scenario 2| Fail|I am finding it a bit confusing when a wallet which is currently being delegated to a pool, but on its epoch+2, it says “Undelegated” and on hover it displays “Redelegate”. From user point of view, it is as if I am redelegating an undelegated pool. Shouldn’t the current pool where the wallet is delegated to get displayed? [See screenshot Scenario 2](https://drive.google.com/file/d/1lXufWia0LEqL1wW2BHvmLPsi48VqD40-/view?usp=sharing) |
|Scenario 8| Pending|Aleksander to check with Nikola and Danilo how to get unknown pool in epoch+2|

**Review by yakovkaravelov — APPROVED** [review]
LGTM! Good job.

File: source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss [hunk 1]
```diff
@@ -17,7 +17,32 @@
...
+      flex: 2.5;
+    }
+
+    .rightBodyTitle {
+      display: flex;
+      flex: 1.5;
+      justify-content: flex-end;
+
+      > span:first-child {
+        margin: 0 52px;
```
Comment (nikolaglumac) [inline source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss:34]: Please avoid using margins for text centering 🙏

File: source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss [hunk 2]
```diff
@@ -17,7 +17,32 @@
...
+      }
+
+      > span:nth-child(2) {
+        margin: 0 50px;
```
Comment (nikolaglumac) [inline source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss:38]: Please avoid using margins for text centering 🙏

File: source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss [hunk 3]
```diff
@@ -17,7 +17,32 @@
...
+      }
+
+      > span:last-child {
+        margin-left: 57px;
+        margin-right: 30px;
```
Comment (nikolaglumac) [inline source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss:43]: Please avoid using margins for text centering 🙏

File: source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss [hunk 4]
```diff
@@ -17,7 +17,32 @@
...
+      > span:first-child {
+        margin: 0 52px;
+      }
+
+      > span:nth-child(2) {
+        margin: 0 50px;
+      }
+
+      > span:last-child {
+        margin-left: 57px;
```
Comment (nikolaglumac) [inline source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss:42]: Please avoid using margins for text centering 🙏

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
@DeeJayElly please fix the following issues:

### 1. The "Delegate" action should not "move" the "Undelegated" text to the top
Screenshot 2020-12-15 at 17 16 17

### 2. The epoch number column headers CSS is to fragile. Please use text centering instead of the margins. The text is NOT correctly centered.
Screenshot 2020-12-15 at 17 15 16

### 3. We will need to figure out how to display status of the "Syncing" wallet as it currently looks strange (we need @darko-mijic and @a-rukin for that):
Screenshot 2020-12-15 at 17 12 06
Screenshot 2020-12-15 at 17 12 09

### 4. Wallet row hover UX - You did not implement the hover effect like planned. The "Redelegate" action should show as soon as you hover the wallet row and not only the tile.

### 5. Stake pool tooltip - this is completely missing...

**Comment by DeeJayElly** [comment]
> @DeeJayElly please fix the following issues:
> 
> ### 1. The "Delegate" action should not "move" the "Undelegated" text to the top
> Screenshot 2020-12-15 at 17 16 17
> 
> ### 2. The epoch number column headers CSS is to fragile. Please use text centering instead of the margins. The text is NOT correctly centered.
> Screenshot 2020-12-15 at 17 15 16
> 
> ### 3. We will need to figure out how to display status of the "Syncing" wallet as it currently looks strange (we need @darko-mijic and @a-rukin for that):
> Screenshot 2020-12-15 at 17 12 06
> 
> Screenshot 2020-12-15 at 17 12 09
> 
> ### 4. Wallet row hover UX - You did not implement the hover effect like planned. The "Redelegate" action should show as soon as you hover the wallet row and not only the tile.
> ### 5. Stake pool tooltip - this is completely missing...

1. Implemented according to the design: https://app.zeplin.io/project/57acfaa7436a750316161a8b/screen/5fb3ad61073a7d76ad070698

5. Stake pool tooltip implemented according to the steps. Its visible on currently delegated stakepool on hover : https://docs.google.com/document/d/1Zv7mDFe5pf3nsW-yUc1_7shaEXlsRaHpOZ_oh4wR4vo/edit#heading=h.i53hp2fdsktq

**Comment by nikolaglumac** [comment]
@DeeJayElly we will discuss this with @darko-mijic tomorrow.

File: source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss (hunk 3 above)
Comment (DeeJayElly) [reply source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss]: Fixed

File: source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss (hunk 4 above)
Comment (DeeJayElly) [reply source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss]: Fixed

File: source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss (hunk 2 above)
Comment (DeeJayElly) [reply source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss]: Fixed

File: source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss (hunk 1 above)
Comment (DeeJayElly) [reply source/renderer/app/components/staking/delegation-center/DelegationCenterBody.scss]: Fixed

**Comment by DeeJayElly** [comment]
@nikolaglumac 
All the request changes are added.

Pending items are: 
- properly positioning Tooltip
- optimizing codebase (current solution does the trick, but it needs a bit of refactoring in code logic)

**Comment by mioriohk** [comment]
@DeeJayElly , for build 15708. Only 1 failure. For complete results, please see Test Summary. (Note: As instructed I haven't tested tooltip, hover etc which are still in development)
|Test cases | Results|Note|
|-----------|----|----|
|Scenario 14|Failed|When testing, the newly updated pool in epoch+2 was not reflected in the drop down list and the listed stake pools. (Note: Observe pools CO2P and ANGEL in the videos: [Video 1](https://drive.google.com/file/d/1_RBjgUwSV7Lu1q6HpH8XHlcDlhW9C6Og/view?usp=sharing) and [Video 2](https://drive.google.com/file/d/1oYgkh_neEe3bZ1w0IbaASo6rLZa7RuX1/view?usp=sharing) ). I am unable to reproduce this issue in production|

**Comment by darko-mijic** [comment]
Looks good so far. Need another epoch to test all cases for delegated wallets. QA should test unknown and undelegated cases too. Unknown can probably be tested by switching SMASH servers. It takes time to load all pools. While pools are loading they are unknown.

**Comment by nikolaglumac** [comment]
> Looks good so far. Need another epoch to test all cases for delegated wallets. QA should test unknown and undelegated cases too. Unknown can probably be tested by switching SMASH servers. It takes time to load all pools. While pools are loading they are unknown.

SMASH server change is not available in this PR.
We also don't have the "undelegate" option enabled.

**Review by a-rukin — APPROVED** [review]

**Review by mioriohk — APPROVED** [review]
LGTM 👍

File: source/renderer/app/components/staking/delegation-center/WalletRow.js [hunk 5]
```diff
@@ -178,160 +236,180 @@ export default class WalletRow extends Component<Props> {
...
+                    {futurePendingDelegationStakePool ? (
+                      <PopOver
+                        key="stakePoolTooltip"
+                        placement="auto"
+                        maxWidth={300}
+                        popperOptions={{
+                          strategy: 'fixed',
+                        }}
+                        themeVariables={{
+                          '--rp-pop-over-bg-color':
```
Comment (yakovkaravelov) [inline source/renderer/app/components/staking/delegation-center/WalletRow.js:318]: Can you define this variables object outside of render function and reference it here?

**Review by yakovkaravelov — CHANGES_REQUESTED** [review]
@DeeJayElly Looks good. Just left minor comment.

**Review by ManusMcCole — APPROVED** [review]

File: source/renderer/app/components/staking/delegation-center/WalletRow.js (hunk 5 above)
Comment (DeeJayElly) [reply source/renderer/app/components/staking/delegation-center/WalletRow.js:318]: Fixed

**Review by yakovkaravelov — APPROVED** [review]

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
@DeeJayElly there are still some implementation issues... Let's discuss those on Monday.

This is one example:
issues

**Comment by DeeJayElly** [comment]
> @DeeJayElly there are still some implementation issues... Let's discuss those on Monday.
> 
> This is one example:
> issues

Fixed

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
This is still wrong...

Screenshot 2021-01-12 at 13 09 16

**Review by mioriohk — CHANGES_REQUESTED** [review]
@DeeJayElly , for build 15844. Only 1 failure observed so far. For complete results, please see Test Summary
|Test cases | Results|Note|
|-----------|----|----|
|Scenario 3| Failed| When the wallet has never been delegated, upon hovering, the button text is 'Redelegate' instead of 'Delegate'. [See screenshot](https://drive.google.com/file/d/1MFs3z2cev_FVl49o11b_Ksy5y0GPQ1gu/view?usp=sharing)

**Comment by DeeJayElly** [comment]
> @DeeJayElly , for build 15844. Only 1 failure observed so far. For complete results, please see Test Summary
> 
> Test cases	Results	Note
> Scenario 3	Failed	When the wallet has never been delegated, upon hovering, the button text is 'Redelegate' instead of 'Delegate'. [See screenshot](https://drive.google.com/file/d/1MFs3z2cev_FVl49o11b_Ksy5y0GPQ1gu/view?usp=sharing)

@mioriohk New updates are pushed. Can you pls recheck?

**Comment by darko-mijic** [comment]
@DeeJayElly, please add screenshots with the stake pool details.

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
It is not possible to move the cursor into the tooltip and copy the stake pool id:
Screenshot 2021-01-12 at 17 06 39

...this is how it should work like:
Screenshot 2021-01-12 at 17 06 25

**Comment by DeeJayElly** [comment]
> @DeeJayElly, please add screenshots with the stake pool details.

Added @darko-mijic

**Comment by mioriohk** [comment]
@DeeJayElly Looks good on build 15866 👍

**Review by mioriohk — APPROVED** [review]

**Review by a-rukin — CHANGES_REQUESTED** [review]
fix tooltip padding 
and popup jumping

**Comment by DeeJayElly** [comment]
> fix tooltip padding
> and popup jumping

Fixed @a-rukin pls recheck it.

**Comment by a-rukin** [comment]
https://user-images.githubusercontent.com/2539345/104566518-c069b600-565e-11eb-8baf-960faea93644.mp4

jumping is still here

**Comment by DeeJayElly** [comment]
>  screencast.2021-01-14.11-50-11.mp4 
> jumping is still here

Fixed

**Comment by a-rukin** [comment]
image
Wondering if it is OK to hide "button" when we are on a popup. Logically for me it's still a part of hovered row.
Any thoughts @darko-mijic @nikolaglumac

**Comment by DeeJayElly** [comment]
> It is not possible to move the cursor into the tooltip and copy the stake pool id:
> Screenshot 2021-01-12 at 17 06 39
> 
> ...this is how it should work like:
> Screenshot 2021-01-12 at 17 06 25

Fixed

**Comment by gnpf** [comment]
@DeeJayElly I found a small issue on build 15888:

- When the stake pool is "Unkown", the alignment on the tile is not centered. Check [this screenshot.](https://drive.google.com/file/d/1wNhYTkgi2GS4u6_I0Dt4dLvgc2tPwSsg/view?usp=sharing)

**Comment by DeeJayElly** [comment]
> @DeeJayElly I found a small issue on build 15888:
> 
> * When the stake pool is "Unkown", the alignment on the tile is not centered. Check [this screenshot.](https://drive.google.com/file/d/1wNhYTkgi2GS4u6_I0Dt4dLvgc2tPwSsg/view?usp=sharing)
> 
> Edit: I found an additional issue:
> 
> * Were all the tooltips updated on this PR? they look different, [check this screenshot.](https://drive.google.com/file/d/1XhGkpjLfmN1jPFH5Rtb5wls2cccQbD4M/view?usp=sharing)
Fixed

**Comment by gnpf** [comment]
@DeeJayElly the fixes for the issues on my previous comment look great.
A few comments about 15892:

- When the tile is not displayed completely and you hover it, an additional scrollbar is displayed and the pop-over is cropped on the bottom. [Check this screenshot.](https://drive.google.com/file/d/16qC3Q0I61HR5FamP7rMsIEd80jx9PKWU/view?usp=sharing)

- Related to the previous item, I was wondering if the pop-over should be displayed from the top when there is no enough space to show it. [This is how it's implemented on the tile view on "Stake pools"](https://drive.google.com/file/d/1lUyaGF970Jifzatu_MpHauyOmvZkDecb/view?usp=sharing)

**Comment by nikolaglumac** [comment]
> * When the tile is not displayed completely and you hover it, an additional scrollbar is displayed and the pop-over is cropped on the bottom. [Check this screenshot.](https://drive.google.com/file/d/16qC3Q0I61HR5FamP7rMsIEd80jx9PKWU/view?usp=sharing)

@DeeJayElly please fix this one ☝️ 

> * Related to the previous item, I was wondering if the pop-over should be displayed from the top when there is no enough space to show it. [This is how it's implemented on the tile view on "Stake pools"](https://drive.google.com/file/d/1lUyaGF970Jifzatu_MpHauyOmvZkDecb/view?usp=sharing)

@gnpf no .- we always need to show the popup on the left side! @DeeJayElly please ignore this one...

**Comment by DeeJayElly** [comment]
> @DeeJayElly the fixes for the issues on [my previous comment](https://github.com/input-output-hk/daedalus/pull/2275#issuecomment-760209481) look great.
> A few comments about 15892:
> 
> * When the tile is not displayed completely and you hover it, an additional scrollbar is displayed and the pop-over is cropped on the bottom. [Check this screenshot.](https://drive.google.com/file/d/16qC3Q0I61HR5FamP7rMsIEd80jx9PKWU/view?usp=sharing)
> * Related to the previous item, I was wondering if the pop-over should be displayed from the top when there is no enough space to show it. [This is how it's implemented on the tile view on "Stake pools"](https://drive.google.com/file/d/1lUyaGF970Jifzatu_MpHauyOmvZkDecb/view?usp=sharing)

@gnpf I have removed the scrollbar.

**Comment by DeeJayElly** [comment]
All the pending issues were fixed for Tile view delegation center. Other tooltips on Stake Pools will be addressed on a separate ticket: [DDW-544](https://jira.iohk.io/browse/DDW-544)

**Comment by a-rukin** [comment]
what about cursor styles? On the "stake pools" page we have different cursor styles, on the "delegation center" only "pointer" style

https://user-images.githubusercontent.com/2539345/104705404-e9577d00-572a-11eb-9f75-3916a45af6e4.mp4

**Review by a-rukin — APPROVED** [review]
LGTM

**Comment by DeeJayElly** [comment]
> what about cursor styles? On the "stake pools" page we have different cursor styles, on the "delegation center" only "pointer" style
> 
>  screencast.2021-01-15.12-11-23.mp4

Added

**Review by gnpf — APPROVED** [review]
Great work 🎉

**Pull request merged**